universum/modules/structure_handler.py

```diff
@@ -66,10 +67,23 @@ def is_successful(self) -> bool:
 class BackgroundStepInfo(TypedDict):
     name: str
     block: Block
-    finalizer: Callable[[], None]
+    process: RunningStepBase
     is_critical: bool
 
 
+class RunningStepBase(ABC):
+
+    error: Optional[str] = None
```

Comment:
The class and instance variables of abcs are not checked by mypy.
So it is better to use some explicit method, like 'get_error', for returning errors.